dataclass_factory/factory.py

```diff
@@ -38,49 +39,56 @@ def serializer(self, class_: Type):
             self.stack.pop()
 
 
+T = TypeVar("T")
+
+
 class Factory:
+    __slots__ = ("default_schema", "debug_path", "schemas")
+
     def __init__(self,
                  default_schema: Optional[Schema] = None,
                  schemas: Optional[Dict[Type, Schema]] = None,
                  debug_path: bool = False):
         self.default_schema = merge_schema(default_schema, DEFAULT_SCHEMA)
         self.debug_path = debug_path
-        self.schemas: Dict[Type, Schema] = defaultdict(lambda: copy(self.default_schema))
+        self.schemas: Dict[Type, Schema] = {}
         if schemas:
             self.schemas.update({
                 type_: merge_schema(schema, self.default_schema)
                 for type_, schema in schemas.items()
             })
 
-    def schema(self, class_: Type) -> Schema:
+    def schema(self, class_: Type[T]) -> Schema:
         schema = self.schemas.get(class_)
         if not schema:
-            schema = copy(self.default_schema)
+            schema = Schema(
+                *astuple(self.default_schema)
+            )
             self.schemas[class_] = schema
         return schema
 
-    def parser(self, class_: Type) -> Parser:
+    def parser(self, class_: Type[T]) -> Parser[T]:
         return self._parser_with_stack(class_, StackedFactory(self))
 
-    def _parser_with_stack(self, class_: Type, stacked_factory: StackedFactory) -> Parser:
+    def _parser_with_stack(self, class_: Type[T], stacked_factory: StackedFactory) -> Parser:
```

Comment:
This line is inconsistent with the previous one: `def parser(self, class_: Type[T]) -> Parser[T]:`

Reply:
But it looks like totally the same.